storage/tree/node_id2.go

```diff
@@ -37,10 +37,28 @@ func NewNodeID2(path string, bits uint) NodeID2 {
 	} else if mx := uint(len(path)) * 8; bits > mx {
 		panic(fmt.Sprintf("NewNodeID2: bits %d > %d", bits, mx))
 	}
-	bytes, tail, mask := split(bits)
-	last := path[bytes] & mask
+	bytes, tailBits := split(bits)
 	// Note: Getting the substring is cheap because strings are immutable in Go.
-	return NodeID2{path: path[:bytes], last: last, bits: tail}
+	return newMaskedNodeID2(path[:bytes], path[bytes], tailBits)
+}
+
+// NewNodeID2WithLast creates a NodeID2 from the given path bytes and the
+// additional last byte, of which only the specified number of most significant
+// bits is used. The number of bits must be between 1 and 8, and can be 0 only
+// if the path bytes string is empty; otherwise the function panics.
+func NewNodeID2WithLast(path string, last byte, bits uint8) NodeID2 {
+	if bits > 8 {
+		panic(fmt.Sprintf("NewNodeID2WithLast: bits %d > 8", bits))
+	} else if bits == 0 && len(path) != 0 {
+		panic("NewNodeID2WithLast: bits=0, but path is not empty")
+	}
+	return newMaskedNodeID2(path, last, bits)
+}
+
+// newMaskedNodeID2 constructs a NodeID ensuring its invariants are met.
```

Comment:
Clarify what happens to `last`.  Is it masked? forwarded untouched to the struct?

Reply:
Yes, it's masked. Done.